Add Vue SDK custom `<UserProfile />` and `<OrganizationProfile />` pages examples

wobsoriano opened the pull request:
### 🔎 Previews:

- https://clerk.com/docs/pr/2002/customization/user-profile
- https://clerk.com/docs/pr/2002/customization/organization-profile

### What does this solve?

- Since the implementation uses Vue [slots](https://vuejs.org/guide/components/slots) (rather than props like in React) to pass in custom components, it's important to document this approach.

### What changed?

- Adds Vue examples to each section of the custom pages docs
- Shows proper usage of template slots for custom components

### Checklist

- [x] I have clicked on "Files changed" and performed a thorough self-review
- [x] I have added the "deploy-preview" label and added the preview link(s) to this PR description
- [x] All existing checks pass

alexisintech commented:
hey! I've updated the copy of both guides, and fixed some of the vue examples.
I've left some TODOs, there are some broken examples that I can't fix on my side - think it needs a source code fix

wobsoriano commented:
> hey! I've updated the copy of both guides, and fixed some of the vue examples. I've left some TODOs, there are some broken examples that I can't fix on my side - think it needs a source code fix

Noted on this, thank you!

alexisintech reviewed (approved):
everything's working!! 😸💖

wobsoriano commented:
> everything's working!! 😸💖

awesomeee!
